examples/gno.land/r/stefann/home/home.gno

```diff
@@ -0,0 +1,274 @@
+		{Name: "Rothenburg", URL: "https://i.ibb.co/cr8d2rQ/rothenburg.jpg"},
+		{Name: "Capetown", URL: "https://i.ibb.co/bPGn0v3/capetown.jpg"},
+		{Name: "Sydney", URL: "https://i.ibb.co/TBNzqfy/sydney.jpg"},
+		{Name: "Oeschinen Lake", URL: "https://i.ibb.co/QJQwp2y/oeschinen-lake.jpg"},
+		{Name: "Barra Grande", URL: "https://i.ibb.co/z4RXKc1/barra-grande.jpg"},
+		{Name: "London", URL: "https://i.ibb.co/CPGtvgr/london.jpg"},
+	}
+	currentCityIndex = 0
+	jarLink = "https://TODO"
+	maxSponsors = 5
+	aboutMe = [2]string{
+		`<h3 style="font-size: 1.4em;">About Me</h3>
+        <p style="font-size: 1.1em;">Hey there! I’m Stefan, a student of Computer Science. I’m all about exploring and adventure — whether it’s diving into the latest tech or discovering a new city, I’m always up for the challenge!</p>`,
+		`<h3 style="font-size: 1.4em;">Contributions</h3>
+        <p style="font-size: 1.1em;">I'm just getting started, but you can follow my journey through Gno.land right here <a href="https://github.com/gnolang/hackerspace/issues/94" target="_blank">🔗</a></p>`,
+	}
+}
+
+func UpdateMaxSponsors(newMax int) {
+	config.AssertAuthorized()
+	maxSponsors = newMax
+}
+
+func UpdateCities(newCities []City) {
+	config.AssertAuthorized()
+	cities = newCities
+}
+
+func UpdateJarLink(newLink string) {
+	config.AssertAuthorized()
+	jarLink = newLink
+}
+
+func UpdatePFP(url, caption string) {
+	config.AssertAuthorized()
+	pfp = url
+}
+
+func UpdateAboutMe(col1, col2 string) {
+	config.AssertAuthorized()
+	aboutMe[0] = col1
+	aboutMe[1] = col2
+}
+
+func Donate() {
+	address := std.GetOrigCaller()
+	amount := std.GetOrigSend()
+
+	if amount.AmountOf("ugnot") == 0 {
+		panic("Donation must include GNOT")
+	}
+
+	found := false
+
+	for i, sponsor := range sponsors {
+		if sponsor.Address == address {
+			sponsors[i].Amount = sponsors[i].Amount.Add(amount)
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		sponsors = append(sponsors, Sponsor{Address: address, Amount: amount})
+	}
+
+	totalDonated = totalDonated.Add(amount)
+
+	totalDonations++
+
+	sortSponsorsByAmount()
+
+	if len(cities) > 0 {
+		currentCityIndex++
+		if currentCityIndex >= len(cities) {
+			currentCityIndex = 0
+		}
+	}
+}
+
+type SponsorSlice []Sponsor
+
+func (s SponsorSlice) Len() int {
+	return len(s)
+}
+
+func (s SponsorSlice) Less(i, j int) bool {
+	return s[i].Amount.AmountOf("ugnot") > s[j].Amount.AmountOf("ugnot")
+}
+
+func (s SponsorSlice) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
+func sortSponsorsByAmount() {
+	sort.Sort(SponsorSlice(sponsors))
+}
+
+func GetTopSponsors() []Sponsor {
+	return sponsors
+}
+
+func CollectDonations() {
+	config.AssertAuthorized()
+
+	banker := std.GetBanker(std.BankerTypeRealmSend)
+
+	ownerAddr := config.Address()
+	banker.SendCoins(std.GetOrigPkgAddr(), ownerAddr, banker.GetCoins(std.GetOrigPkgAddr()))
+}
+
+func GetTotalDonations() std.Coins {
+	return totalDonated
+}
+
+func GetDonationCount() int {
+	return totalDonations
+}
```

Comment:
I am not sure if this addresses the issue you have in mind, but I removed `TotalDonations` as it can be derived from the `sponsors` map directly. I also grouped `TotalDonations` (now renamed to `DonationsCount`) with sponsors in a shared object. (commit: 139c8d3a5fc245af555c1c2558a4e0eb2a1b94e3)